lib/rubocop/cop/rspec/describe_class.rb

```diff
@@ -27,36 +40,42 @@ class DescribeClass < Base
         MSG = 'The first argument to describe should be '\
               'the class or module being tested.'
 
-        def_node_matcher :rails_metadata?, <<-PATTERN
-          (pair
-            (sym :type)
-            (sym { :channel :controller :helper :job :mailer :model :request
-                   :routing :view :feature :system :mailbox })
-          )
-        PATTERN
-
-        def_node_matcher :example_group_with_rails_metadata?, <<~PATTERN
-          (send #rspec? :describe ... (hash <#rails_metadata? ...>))
+        def_node_matcher :example_group_with_ignored_metadata?, <<~PATTERN
+          (send #rspec? :describe ... (hash <#ignored_metadata? ...>))
```

Comment:
Speaking of `describe`, there are more than just it:
```ruby
        GROUPS  = SelectorSet.new(%i[describe context feature example_group])
        SKIPPED = SelectorSet.new(%i[xdescribe xcontext xfeature])
        FOCUSED = SelectorSet.new(%i[fdescribe fcontext ffeature])
```

If you change the entry point from `on_top_level_group` to `on_top_level_example_group`, you'll be skipping shared example groups.

In this case, if you want to remove the limitation of only detecting `describe`, it's as easy as removing `:describe` part from the node pattern above + a spec example that e.g. `context` is flagged, but `shared_examples_for` are not.

Can be handled separately. Just let me know what you'll decide, I can file a separate ticket.

Reply:
There's one more place to fix, `not_a_const_described`:
```
(send #rspec? :describe $[!const !#string_constant?] ...)
```

Reply:
Filed https://github.com/rubocop-hq/rubocop-rspec/issues/1025